File: Checks/PWR005/example-omp.c
// PWR005: Disable default OpenMP scoping

void example(int *result, unsigned size) {
  // NOT-PWR002: Declaration outside loop to trigger PWR005
  int t;

  // Default data scoping is used, making `t` shared instead of private
  // NOT-PWR004: omitting scoping specifiers to highlight dangers of PWR005
  #pragma omp parallel for
  for (int i = 0; i < size; i++) {
    t = i + 1;
    result[i] = t;
  }
}

File: Checks/PWR005/example-omp.f90
! PWR005: Disable default OpenMP scoping

subroutine example(result)
  implicit none
  integer, intent(out) :: result(:)
  integer :: i, t

  ! Default data scoping is used, making `t` shared instead of private
  ! NOT-PWR004: omitting scoping specifiers to highlight dangers of PWR005
  !$omp parallel do
  do i = 1, size(result, 1)
    t = i + 1
    result(i) = t
  end do
  !$omp end parallel do
end subroutine example
